_ml-commons-plugin/agents-tools/tools/ppl-tool.md

```diff
@@ -0,0 +1,206 @@
+      "parameters": {
+        "model_id": "h5AUWo0BkIylWTeYT4SU",
+        "model_type": "FINETUNE",
+        "execute": true
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %} 
+
+For parameter descriptions, see [Register parameters](#register-parameters).
+
+OpenSearch responds with an agent ID:
+
+```json
+{
+  "agent_id": "9X7xWI0Bpc3sThaJdY9i"
+}
+```
+
+## Step 4: Run the agent
+
+Before you run the agent, make sure that you add the sample OpenSearch Dashboards `Sample web logs` dataset. To learn more, see [Adding sample data]({{site.url}}{{site.baseurl}}/dashboards/quickstart#adding-sample-data).
+
+Then, run the agent by sending the following request:
+
+```json
+POST /_plugins/_ml/agents/9X7xWI0Bpc3sThaJdY9i/_execute
+{
+  "parameters": {
+    "verbose": true,
+    "question": "what is the error rate yesterday",
+    "index": "opensearch_dashboards_sample_data_logs"
+  }
+}
+```
+{% include copy-curl.html %} 
+
+OpenSearch returns the PPL query and the query results:
+
+```json
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "response",
+          "result":"{\"ppl\":\"source\=opensearch_dashboards_sample_data_logs| where timestamp \> DATE_SUB(NOW(), INTERVAL 1 DAY) AND timestamp \< NOW() | eval is_error\=IF(response\=\'200\', 0, 1.0) | stats AVG(is_error) as error_rate\",\"executionResult\":\"{\\n  \\\"schema\\\": [\\n    {\\n      \\\"name\\\": \\\"error_rate\\\",\\n      \\\"type\\\": \\\"double\\\"\\n    }\\n  ],\\n  \\\"datarows\\\": [\\n    [\\n      null\\n    ]\\n  ],\\n  \\\"total\\\": 1,\\n  \\\"size\\\": 1\\n}\"}"
+        }
+      ]
+    }
+  ]
+}
+```
+
+If you set `execute` to `false`, OpenSearch only returns the query but does not run it:
+
+```json
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+            "name": "response",
+            "result": "source=opensearch_dashboards_sample_data_logs| where timestamp > DATE_SUB(NOW(), INTERVAL 1 DAY) AND timestamp < NOW() | eval is_error=IF(response='200', 0, 1.0) | stats AVG(is_error) as error_rate"
+        }
+      ]
+    }
+  ]
+}
+```
+
+## Register parameters
+
+The following table lists all available parameters when registering the tool.
+
+Parameter	| Type | Required/Optional | Description	
+:--- | :--- | :--- | :---
+`model_id` | String | Required | The model ID of the large language model (LLM) to use for translating text into a PPL query.
+`model_type` | String | Optional | The model type. Valid values are `CLAUDE` (Anthropic Claude model), `OPENAI` (OpenAI models), and `FINETUNE` (custom fine-tuned model). 
+`prompt` | String | Optional | The prompt to provide to the LLM.
+`execute` | Boolean | Optional | Specifies whether to run the PPL query. Default is `true`.
+`input` | Object | Optional | Contains two parameters that specify the index to search and the question for the LLM. For example, `"input": "{\"index\": \"${parameters.index}\", \"question\": ${parameters.question} }"`.
```

Comment:
One more parameter can be config here: `head` Int parameter, will limit the return number of PPL execution result if customer set execute to true. Default is -1, which means not truncation.